### DISCUSSION TITLE
`use enum E ...`

### DISCUSSION THREAD

**Discussion by mlutze** [flix/flix#5632] [Ideas]
The new requirement to qualify enum cases is principled, but it feels awkward to `use` something _above_ its declaration site. I'm wondering if we can have an opt-in syntax to automatically `use` all the cases in an enum. For example:

```
use pub enum MyList[a] {
    case MyNil
    case MyCons(a, MyList[a])
}
```

which is effectively

```
use MyList.{MyNil, MyCons}

pub enum MyList[a] {
    case MyNil
    case MyCons(a, MyList[a])
}
```

**Comment by magnus-madsen** [comment]
> The new requirement to qualify enum cases is principled.

Yes, and it is consistent with Rust. I think it teaches you to use the module some appropriately.

(Transferring to discussion.)
